### PR TITLE
fix(TPG >= 4.68)!: added missing features for egress policies

### PR DESCRIPTION
**Pull request by cmalpe**

### CODE REVIEW

File: modules/regular_service_perimeter/main.tf [hunk 1]
```diff
@@ -76,6 +76,13 @@ resource "google_access_context_manager_service_perimeter" "regular_service_peri
         egress_from {
           identity_type = lookup(egress_policies.value["from"], "identity_type", null)
           identities    = lookup(egress_policies.value["from"], "identities", null)
+          dynamic "sources" {
```
Comment (imrannayer) [inline modules/regular_service_perimeter/main.tf:79]: @cmalpe can u plz use the logic used in ingress_from.sources?

File: modules/regular_service_perimeter/main.tf [hunk 2]
```diff
@@ -161,6 +168,13 @@ resource "google_access_context_manager_service_perimeter" "regular_service_peri
           egress_from {
             identity_type = lookup(egress_policies.value["from"], "identity_type", null)
             identities    = lookup(egress_policies.value["from"], "identities", null)
+            dynamic "sources" {
```
Comment (imrannayer) [inline modules/regular_service_perimeter/main.tf:171]: @cmalpe can u plz use the logic used in ingress_policies.ingress_from.sources?

File: modules/regular_service_perimeter/main.tf [hunk 3]
```diff
@@ -28,13 +28,15 @@ resource "google_access_context_manager_service_perimeter" "regular_service_peri
 
   status {
     restricted_services = var.restricted_services
+    resources           = [for item in var.resources_dry_run : can(regex("global/networks", item)) ? format("//compute.googleapis.com/%s", item) : format("projects/%s", item)]
```
Comment (imrannayer) [inline modules/regular_service_perimeter/main.tf:31]: @cmalpe it should be `var.resources` instead of `var.resources_dry_run`.

Comment (cmalpe) [reply modules/regular_service_perimeter/main.tf]: Hi @imrannayer I missed that part! Thanks for finding this error. I have corrected it and pushed the changes.

File: modules/regular_service_perimeter/main.tf [hunk 4]
```diff
@@ -28,13 +28,15 @@ resource "google_access_context_manager_service_perimeter" "regular_service_peri
 
   status {
     restricted_services = var.restricted_services
+    resources           = [for item in var.resources : can(regex("global/networks", item)) ? format("//compute.googleapis.com/%s", item) : format("projects/%s", item)]
```
Comment (g-awmalik) [inline modules/regular_service_perimeter/main.tf:31]: I think this is already adding the resource i.e. the project to the perimeter and then `google_access_context_manager_service_perimeter_resource` tries to add it again leading to:

```
STDERR: Error: Unable to create ServicePerimeterResource, existing object already found: map[resource:projects/838237599229]

  with module.example.module.regular_service_perimeter_1.google_access_context_manager_service_perimeter_resource.service_perimeter_resource["838237599229"],
  on ../../../modules/regular_service_perimeter/main.tf line 229, in resource "google_access_context_manager_service_perimeter_resource" "service_perimeter_resource":
 229: resource "google_access_context_manager_service_perimeter_resource" "service_perimeter_resource" {
---- End output of terraform apply -auto-approve -lock=true -lock-timeout=0s -input=false -no-color -paral
```

Comment (imrannayer) [reply modules/regular_service_perimeter/main.tf:31]: @cmalpe can u plz remove resource = ....

Comment (cmalpe) [reply modules/regular_service_perimeter/main.tf]: This change is done @imrannayer

**Pull request merged**